### PR TITLE
fix(condo): DOMA-9882 test for sending news to the right resident

### PR DESCRIPTION
**Pull request by tolmachev21**
Test for the latest prodaction bug sending messages to wrong users

### CODE REVIEW

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 1]
```diff
@@ -243,6 +243,80 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+            const messageWhere2 = { user: { id: residentClient2.user.id }, type: NEWS_ITEM_COMMON_MESSAGE_TYPE }
+
+            // Publish NewsItem to make it sendable
+            await publishTestNewsItem(adminClient, newsItem.id)
+            await notifyResidentsAboutNewsItem(newsItem.id)
+
+            // The message should not be sent to the second user with a different address, but the same unitName and unitType 
+            await waitFor(async () => {
+                const messages1 = await Message.getAll(adminClient, messageWhere1)
+                const messages2 = await Message.getAll(adminClient, messageWhere2)
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:310]: you can use `.getOne` here and check `expect(message1).toBeDefined()` and `expect(message2).toBeUndefined()` below

**Review by nomerdvadcatpyat — APPROVED** [review]

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js (hunk 1 above)
Comment (tolmachev21) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:310]: Okey, I am use getOne, but message2 receives [], which will not pass the test toBeUndefinded()

Comment (tolmachev21) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:310]: Ow, method getAll return only object, not array. Okey, I will learn it, thanks^

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js [hunk 2]
```diff
@@ -243,6 +243,77 @@ describe('notifyResidentsAboutNewsItem', () => {
...
+            await publishTestNewsItem(adminClient, newsItem.id)
+            await notifyResidentsAboutNewsItem(newsItem.id)
+
+            // The message should not be sent to the second user with a different address, but the same unitName and unitType 
+            await waitFor(async () => {
+                const messages1 = await Message.getOne(adminClient, messageWhere1)
+                const messages2 = await Message.getOne(adminClient, messageWhere2)
+
+                expect(messages1).toBeDefined()
+                expect(messages2).toBeUndefined()
```
Comment (toplenboren) [inline apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:313]: There might be a case when:

1. messages1 is successfully sent (it is defined)
2. messages2 is GOING to be sent, but the sending process just havent started yet (it is undefined)

In this case your test is going to be successful

Comment (toplenboren) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:313]: But in essesnse it should fail

Comment (Alllex202) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:313]: Yes, it is

Do you have any idea how to solve this case?

Comment (Alllex202) [reply apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js:313]: As a suggestion, we can in the function of sending news count the number of created messages and return this value. And check this count in test

**Pull request merged**